### ISSUE TITLE
Basic AspNetCore service doesn't work with AllowSynchronousIO set to false

### ISSUE THREAD

**Issue by mikedld** [CXuesong/JsonRpc.Standard#13]
Presumably since dotnet/aspnetcore#7644, basic service (similar to that in #12) is failing with:

```
info: Microsoft.AspNetCore.Hosting.Diagnostics[1]
      Request starting HTTP/1.1 POST http://127.0.0.1:32900/jsonapi application/json 87
fail: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[1]
      An unhandled exception has occurred while executing the request.
System.InvalidOperationException: Synchronous operations are disallowed. Call WriteAsync or set AllowSynchronousIO to true instead.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Dispose(Boolean disposing)
   at System.IO.TextWriter.Dispose()
   at JsonRpc.AspNetCore.AspNetCoreRpcServerHandler.ProcessRequestAsync(HttpContext context)
   at JsonRpc.AspNetCore.JsonRpcAspNetExtensions.<>c__DisplayClass1_0.<<UseJsonRpc>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at AppOptics.Instrumentation.AppOpticsMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
warn: Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware[2]
      The response has already started, the error page middleware will not be executed.
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      Connection id "0HM0EHOLKK1NQ", Request id "0HM0EHOLKK1NQ:00000001": An unhandled exception was thrown by the application.
System.InvalidOperationException: Synchronous operations are disallowed. Call WriteAsync or set AllowSynchronousIO to true instead.
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpResponseStream.Write(Byte[] buffer, Int32 offset, Int32 count)
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Dispose(Boolean disposing)
   at System.IO.TextWriter.Dispose()
   at JsonRpc.AspNetCore.AspNetCoreRpcServerHandler.ProcessRequestAsync(HttpContext context)
   at JsonRpc.AspNetCore.JsonRpcAspNetExtensions.<>c__DisplayClass1_0.<<UseJsonRpc>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at AppOptics.Instrumentation.AppOpticsMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddleware.Invoke(HttpContext context)
   at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
info: Microsoft.AspNetCore.Hosting.Diagnostics[2]
      Request finished in 149.2918ms 500 application/json-rpc
```

Currently working around with https://github.com/dotnet/aspnetcore/issues/7644#issuecomment-538162893.

**Comment by CXuesong** [comment]
Check the followng part in your stack trace
```cs
   at System.IO.StreamWriter.Flush(Boolean flushStream, Boolean flushEncoder)
   at System.IO.StreamWriter.Dispose(Boolean disposing)
   at System.IO.TextWriter.Dispose()
   at JsonRpc.AspNetCore.AspNetCoreRpcServerHandler.ProcessRequestAsync(HttpContext context)
```

I think using `DisposeAsync` should resolve the problem.

**Comment by CXuesong** [comment]
Released fix in [v0.5.4](https://github.com/CXuesong/JsonRpc.Standard/releases/tag/v0.5.4).

**Comment by mikedld** [comment]
Thanks for a swift fix!
